### PR TITLE
Distinguish between App Store and Ad Hoc profiles

### PR DESCRIPTION
**Pull request by godrei**

### CODE REVIEW

File: lib/autoprovision/portal/profile_client.rb [hunk 1]
```diff
@@ -136,6 +125,28 @@ def self.ensure_manual_profile(certificate, app, distribution_type, allow_retry
...
     private_class_method
 
+    def self.fetch_profiles(distribution_type, xcode_managed)
+      profile_class = portal_profile_class(distribution_type)
+
+      profiles = []
+      run_and_handle_portal_function { profiles = profile_class.all(mac: false, xcode: xcode_managed) }
+
+      profiles = profiles.select(&:managed_by_xcode?) if xcode_managed
+      profiles = profiles.reject { |profile| profile.sub_platform.to_s == 'tvOS' }
```
Comment (viktorbenei) [inline lib/autoprovision/portal/profile_client.rb]: actually, we should have an option for tvOS profiles too, right?

**Comment by godrei** [comment]
@yonaskolb lowercased compare is a good idea. what do you mean by 'more generic way'?

**Comment by yonaskolb** [comment]
I just meant future proofing it so instead of checking for tvOS specifically it would just make sure the lowercased platforms matched. (One might be osx vs macOS depending on the naming though...)
What you have is fine for now.

File: lib/autoprovision/certificate_helper.rb [hunk 2]
```diff
@@ -87,15 +87,19 @@ def ensure_certificate(name, team_id, distribution_type)
     end
 
     if distribution_type == 'development' && @development_certificate_info.nil?
-      raise "Selected distribution type: development, but forgot to provide a Development type certificate.\n" \
-  "Don't worry, it's really simple to fix! :)\n" \
-  "Simply provide a Development type certificate (.p12) and we'll be building in no time!"
+      raise [
+        'Selected distribution type: development, but forgot to provide a Development type certificate.',
+        "Don't worry, it's really simple to fix! :)",
+        "Simply provide a Development type certificate (.p12) and we'll be building in no time!"
```
Comment (trapacska) [inline lib/autoprovision/certificate_helper.rb]: I don't want this line a lot longer but please include a very short part of the how to like: 
`Simply provide a Development type certificate (.p12) and we'll be building in no time!` -> 
`Simply upload Development type certificate (.p12) on the workflow editor's CodeSign tab`

File: lib/autoprovision/certificate_helper.rb [hunk 3]
```diff
@@ -87,15 +87,19 @@ def ensure_certificate(name, team_id, distribution_type)
...
     end
 
     if distribution_type != 'development' && @production_certificate_info.nil?
-      raise "Selected distribution type: #{distribution_type}, but forgot to provide a Distribution type certificate.\n" \
-  "Don't worry, it's really simple to fix! :)\n" \
-  "Simply provide a Distribution type certificate (.p12) and we'll be building in no time!"
+      raise [
+        "Selected distribution type: #{distribution_type}, but forgot to provide a Distribution type certificate.",
+        "Don't worry, it's really simple to fix! :)",
+        "Simply provide a Distribution type certificate (.p12) and we'll be building in no time!"
```
Comment (trapacska) [inline lib/autoprovision/certificate_helper.rb]: same as above

File: lib/autoprovision/portal/profile_client.rb [hunk 4]
```diff
@@ -136,6 +127,34 @@ def self.ensure_manual_profile(certificate, app, distribution_type, allow_retry
 
     private_class_method
 
+    def self.fetch_profiles(distribution_type, xcode_managed, platform)
+      profile_class = portal_profile_class(distribution_type)
+
+      profiles = []
+      run_and_handle_portal_function { profiles = profile_class.all(mac: false, xcode: xcode_managed) }
```
Comment (trapacska) [inline lib/autoprovision/portal/profile_client.rb]: is mac: always false? So is there any platform that might change it to true or smtg?

**Pull request merged**